dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/runner/TaskExecuteThread.java

```diff
@@ -191,14 +197,25 @@ public void run() {
             this.task.init();
             //init varPool
             this.task.getParameters().setVarPool(taskExecutionContext.getVarPool());
-            // task handle
-            this.task.handle();
 
-            // task result process
-            if (this.task.getNeedAlert()) {
-                sendAlert(this.task.getTaskAlertInfo());
+            taskInstanceMapper = SpringApplicationContext.getBean(TaskInstanceMapper.class);
+            TaskInstance taskInstance = taskInstanceMapper.selectById(taskExecutionContext.getTaskInstanceId());
```

Comment:
You can use the method "ProcessService.findTaskInstanceById()", no need introduce the TaskInstanceMapper class